numbat/modules/core/numbers.nbt

```diff
@@ -23,4 +23,9 @@ fn is_zero<D: Dim>(value: D) -> Bool = value == 0
 @description("Returns true unless the input is 0 (zero).")
 @example("is_nonzero(37)")
 @example("is_nonzero(0)")
-fn is_nonzero<D: Dim>(value: D) -> Bool = !is_zero(value)
\ No newline at end of file
+fn is_nonzero<D: Dim>(value: D) -> Bool = !is_zero(value)
+
+@description("Returns true if the input is an integer. If the input has units, then returns whether it is an integral number of its units.")
+@example("is_int(30 seconds)")
+@example("is_int(0.5 minutes)")
+fn is_int<D: Dim>(value: D) -> Bool
```

Comment:
I think this should be a function on `Scalar`. If we make it work on dimensionful quantities, it introduces inconsistencies, as you demonstrated in the example:
```
true = is_int(30 seconds) ?=? is_int(0.5 minutes) = false
```
We should avoid this if possible (I know that there are already ways to introduce inconsistencies like this via `value_of`; but those functions are at least marked "unsafe").

Minor: I'd like to rename it to `is_integer`

Also, can we implement it in Numbat, using `fract`?
```
fn is_integer(value: Scalar) -> Bool = is_zero(fract(value))
```

Reply:
Can do. What is the benefit of implementing functions in Numbat over Rust?

Reply:
That is a good question.

I think of the "FFI bridge" between Rust and Numbat as something that helps Numbat to get up on its feet. We will always need support from the "runtime". We will probably never implement our own PRNG to implement `rand` in Numbat. We will almost certainly never have a full networking stack to be able to look up currency exchange rates from within Numbat. And we will never implement `print` in Numbat.

But as Numbat grows into a more mature language of its own, I think it makes sense to keep the foreign-function interface as small as possible. Especially in areas which belong to the core functionality of Numbat like mathematical computations.

Having part of the "standard library" written in Numbat itself helps us to design APIs in a way that makes them easy to use … in Numbat itself. #646 is a good example. If all string functions were written in Rust and re-exported to Numbat, may we wouldn't have found out that they are not as ergonomic as they could be.

The standard library is the biggest Numbat "code base" that we have, and it's a good "benchmark" to have when we make changes to the language.

There are also practical advantages to having functions implemented in Numbat itself. Even if we reimplement something that would be readily available in Rust, it's typically way less code to maintain. The function name appears just once in the `fn …` definition instead of multiple times for a FFI function. It also allows us to change/develop functions without having to recompile the Numbat interpreter.

There's also downsides of course. Performance, for example. If we see that a heavily-used function is too slow, we can think about re-implementing it in Rust.

Side remark: Note that there are also some functions like `map` that can't even be implemented in Rust at the moment, as we don't have a way to call other Numbat functions from FFI functions.